src/librustc_middle/ty/print/pretty.rs

```diff
@@ -1022,7 +1024,7 @@ pub trait PrettyPrinter<'tcx>:
                 )?;
             }
             (Scalar::Ptr(ptr), ty::FnPtr(_)) => {
-                let instance = self.tcx().unwrap_fn(ptr.alloc_id);
+                let instance = self.tcx().get_global_alloc(ptr.alloc_id).unwrap().unwrap_fn();
```

Comment:
Ah I forgot that we now have double-unwraps... hm. Not sure if it is worth trying to do something about them?

Reply:
`self.tcx()[ptr.alloc_id]` would solve it ;)

Reply:
A method `global_alloc(alloc_id)` that panics on not-found would, too. ;)

Reply:
done